server/hotrod/src/main/scala/org/infinispan/server/hotrod/Decoder2x.scala

```diff
@@ -383,9 +383,9 @@ object Decoder2x extends AbstractVersionedDecoder with ServerConstants with Log
                case _ => false
             }
             val reg = server.getClientListenerRegistry
-            reg.addClientListener(ch, h, listenerId, cache, includeState,
+            reg.addClientListener(this, ch, h, listenerId, cache, includeState,
                   (filterFactoryInfo, converterFactoryInfo), useRawData)
-            createSuccessResponse(h, null)
+            null
```

Comment:
This causes NPE in  [HotRodTestingUtil#withClientListener](https://github.com/infinispan/infinispan/blob/master/server/hotrod/src/test/scala/org/infinispan/server/hotrod/test/HotRodTestingUtil.scala#L394) (and thus test failures)